src/ocean/util/StructShallowReset.d

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+
+    Utility to reset fields of a struct to their .init value while preserving
+    array pointers (their length is set to 0 but memory is kept available for
+    further reuse).
+
+    Although the functionality provided in this module is similar to that in
+    `ocean.util.DeepReset`, it differs in some fundamental ways (details of
+    which can be found in the function documentation below).
+
+    This utility is intended to be used in very specific cases, with
+    `ocean.util.DeepReset` being the preferred choice for general resetting
+    needs.
+
+    Copyright: Copyright (c) 2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.util.StructShallowReset;
```

Comment:
For parity with `DeepReset`, it might make more sense to name this `ShallowReset` and internally `static assert` that it only supports structs. (I guess this limitation exists for the sake of simplicity... not having to recurse into fields of super classes?)

Reply:
>I guess this limitation exists for the sake of simplicity

Yep. And the fact that we currently don't need it for anything other than structs.

>For parity with DeepReset, it might make more sense to name this ShallowReset and internally static assert that it only supports structs. 

Yes, that can be done. The main reason the `Struct` was added to the module name was to make it clear from the get-go that only structs are supported (to further discourage someone from using this rather than `DeepReset`). But I guess this will be irrelevant if we go the path of adding a policy argument to `DeepReset` itself.